File: JAVA/2981. Find Longest Special Substring That Occurs Thrice I.java
class Solution {
    public int maximumLength(String s) {
        HashMap<String,Integer>m=new HashMap<>();
        for(int i=0;i<s.length();i++){
            String k="";
            k+=s.charAt(i);
            m.put(k.toString(), m.getOrDefault(k.toString(), 0) + 1);
            for(int j=i+1;j<s.length();j++){
                if(k.charAt(k.length()-1)==s.charAt(j)){
                    k+=s.charAt(j);
                    m.put(k.toString(), m.getOrDefault(k.toString(), 0) + 1);
                }
                else
                break;
            }
        }
        int a=0;
        for(String i:m.keySet()){
            if(m.get(i)>=3&&i.length()>a)
            a=i.length();
        }
        if(a==0)
        return -1;
        return a;
    }
}

File: C++/2981. Find Longest Special Substring That Occurs Thrice I.cpp
class Solution {
public:
    int maximumLength(string s) {
        map<string,int>m;
        for(int i=0;i<s.size();i++){
            string k="";
            k+=s[i];
            m[k]++;
            for(int j=i+1;j<s.size();j++){
                if(k.back()==s[j]){
                k+=s[j];
                m[k]++;
                }
                else
                break;
            }
        }
        int a=0;
        for(auto i:m){
            if(i.second>2&&i.first.size()>a)
            a=i.first.size();
        }
        if(a==0)
        return -1;
        return a;
    }
};